packages/medusa/src/api/store/product-variants/[id]/route.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+  refetchEntity,
+} from "@medusajs/framework/http"
+import { HttpTypes } from "@medusajs/framework/types"
+import { MedusaError, QueryContext } from "@medusajs/framework/utils"
+import { wrapVariantsWithInventoryQuantityForSalesChannel } from "../../../utils/middlewares"
+import { StoreRequestWithContext } from "../../types"
+import { wrapVariantsWithTaxPrices } from "../helpers"
+import { StoreProductVariantParamsType } from "../validators"
+
+type StoreVariantRetrieveRequest =
+  StoreRequestWithContext<HttpTypes.StoreProductVariantParams> &
+    AuthenticatedMedusaRequest<StoreProductVariantParamsType>
+
+export const GET = async (
+  req: StoreVariantRetrieveRequest,
+  res: MedusaResponse<HttpTypes.StoreProductVariantResponse>
+) => {
+  const withInventoryQuantity =
+    req.queryConfig.fields.includes("inventory_quantity")
+
+  if (withInventoryQuantity) {
+    req.queryConfig.fields = req.queryConfig.fields.filter(
+      (field) => field !== "inventory_quantity"
+    )
+  }
+
+  const idOrFilter: Record<string, unknown> = {
+    ...req.filterableFields,
+    id: req.params.id,
+  }
+
+  if (req.pricingContext) {
+    idOrFilter.context = {
+      calculated_price: QueryContext(req.pricingContext),
+    }
+  }
+
+  const variant = await refetchEntity({
```

Comment:
nit(non-blocking): we've talked about gradually moving away from using the `refetchEntity` and `refetchEntities` because they are nothing but a wrapper around `query.graph` call and doesn't add much value. Should we use it here?